frontend/src/components/common/form/TimePicker.tsx

```diff
@@ -0,0 +1,216 @@
+import type { ForwardedRef, ReactElement } from "react";
+import React, { forwardRef, useRef } from "react";
+import { Box, Input, useMergeRefs } from "@chakra-ui/react";
+import type { ChakraStylesConfig, SelectInstance } from "chakra-react-select";
+
+import type { StringOption } from "../../../types/SelectInputTypes";
+
+import Select from "./Select";
+
+const MINUTE_OPTIONS: StringOption[] = [...Array(60).fill(null)].map(
+  (_, num) => ({
+    label: num.toString().padStart(2, "0"),
+    value: num.toString(),
+  }),
+);
+
+const HOUR_OPTIONS: StringOption[] = [...Array(24).fill(null)].map(
+  (_, num) => ({
+    label: num.toString().padStart(2, "0"),
+    value: num.toString(),
+  }),
+);
+
+const chakraStyles = (
+  align: "left" | "right",
+): ChakraStylesConfig<StringOption> => ({
+  control: (provided) => ({
+    ...provided,
+    background: "unset",
+    _hover: {
+      background: "unset",
+    },
+    _focusVisible: {
+      outline: "none",
+    },
+    border: "none",
+  }),
+  valueContainer: (provided) => ({
+    ...provided,
+    p: 0,
+    justifyContent: align === "left" ? "flex-start" : "flex-end",
+    _focusWithin: {
+      color: "rgba(255, 255, 255, 0.7)",
+    },
+    minW: "calc(2em)",
+  }),
+  inputContainer: (provided) => ({
+    ...provided,
+    p: "0.125rem",
+    m: 0,
+    borderRadius: 3,
+    _focusWithin: {
+      bg: "blue.500",
+    },
+    justifyContent: align === "left" ? "flex-start" : "flex-end",
+  }),
+  input: (provided) => ({
+    ...provided,
+    caretColor: "white",
+    color: "white",
+    textAlign: align,
+    cursor: "inherit",
+  }),
+  placeholder: (provided) => ({
+    ...provided,
+  }),
+  indicatorsContainer: (provided) => ({
+    ...provided,
+    display: "none",
+  }),
+  downChevron: (provided) => ({
+    ...provided,
+    display: "none",
+  }),
+  container: (provided) => ({
+    ...provided,
+    flex: 1,
+  }),
+  menu: (provided) => ({
+    ...provided,
+    w: 200,
+  }),
+});
+
+type TimePickerProps = {
+  name?: string;
+  value: Date | null | undefined;
+  defaultDay?: Date;
+  onChange: (date: Date) => void;
+  isDisabled?: boolean;
+  attachmentDirection?: "left" | "right";
+};
+
+const TimePicker = forwardRef(function TimePicker(
+  {
+    name,
+    value,
+    defaultDay,
+    onChange,
+    isDisabled,
+    attachmentDirection,
```

Comment:
are we using this `attachmentDirection`?

Reply:
Good point! This was left over from when I was playing around with designs, and I left it in in case we wanted to go back. I'll remove it now!